libr/anal/sign.c

```diff
@@ -89,6 +89,60 @@ R_API RList *r_sign_fcn_refs(RAnal *a, RAnalFunction *fcn) {
 	return ret;
 }
 
+static RSignBytes *des_bytes_norm(const char *in) {
+	// "444444:ffffff" or "44444444"
+	RSignBytes *b = R_NEW0 (RSignBytes);
+	if (b && (b->size = r_hex_str2bin_until_new (in, &b->bytes)) > 0) {
+		in += 2 * b->size;
+		if (*in == '\0' && (b->mask = malloc (b->size)) != NULL) {
+			// no mask, set it to f's
+			memset (b->mask, 0xff, b->size);
+			return b;
+		} else if (*in++ == ':') {
+			// get mask
+			int size = r_hex_str2bin_until_new (in, &b->mask);
+			in += 2 * size;
+			if (size == b->size && *in == '\0') {
+				return b;
+			}
+		}
+	}
+	r_sign_bytes_free (b);
+	return NULL;
+}
+
+static RSignBytes *deserialize_bytes(const char *in) {
+	RSignBytes *b = des_bytes_norm (in);
+	if (b) {
+		return b;
+	}
+
+	// "44..44.."
+	b = R_NEW0 (RSignBytes);
+	size_t len = strlen (in) + 3;
+	b->bytes = malloc (len);
+	b->mask = malloc (len);
+	if (b && b->bytes && b->mask) {
+		b->size = r_hex_str2binmask (in, b->bytes, b->mask);
+		if (b->size > 0) {
+			return b;
+		}
+	}
+	return NULL;
+}
+
+static RSignBytes *deserialize_anal(RAnal *a, const char *in) {
+	RSignBytes *b = R_NEW0 (RSignBytes);
+	if (b && (b->size = r_hex_str2bin_until_new (in, &b->bytes)) > 0) {
+		in += 2 * b->size;
+		if (*in == '\0' && (b->mask = r_anal_mask (a, b->size, b->bytes, 0))) {
```

Comment:
same !*in

and im not a big fan of assignments in if statements, because that leads to confision and iirc some compilers will warn on that

```suggestion
		if (!*in && (b->mask = r_anal_mask (a, b->size, b->bytes, 0))) {
```